native-schema-registry/c/include/glue_schema_registry_deserializer.h

```diff
@@ -0,0 +1,32 @@
+#ifndef GLUE_SCHEMA_REGISTRY_DESERIALIZER_H
+#define GLUE_SCHEMA_REGISTRY_DESERIALIZER_H
+
+#include "glue_schema_registry_schema.h"
+#include "glue_schema_registry_error.h"
+#include "mutable_byte_array.h"
+#include "read_only_byte_array.h"
+#include <stdbool.h>
+
+typedef struct glue_schema_registry_deserializer {
```

Comment:
not blocking anything here, because it's 100% your call, but the typedef struct idiom goes off the rails when used too frequently. I tend to just use 'struct type' declarations.

Reply:
I will check how it plays with Swig and take a call.

Reply:
It plays well with Swig as Swig generates the class structures in target languages from typefdef structs, https://www.swig.org/Doc4.0/SWIGDocumentation.html#SWIG_nn32
Leaving it as is.